### DISCUSSION TITLE
🧠 Kairos presentations brainstorming

### DISCUSSION THREAD

**Discussion by mauromorales** [kairos-io/kairos#1872] [General]
Part of the community work we want to do with Kairos, is to reach individuals and other communities, so they can benefit from our project, and so we can benefit from their feedback. One way to do this, is by giving talks about the Kairos project and the problems it solves.

This year the Kairos team introduced Kairos at the following events: FOSDEM, KCD Amsterdam, KCD Paris and DevConf. While this outreach was good for our first year as a team together, I would like us to be more prepared and try to have even more impact on 2024. For this reason, I've created the [🛫 Outreach](https://github.com/orgs/kairos-io/projects/4) project for us to track events where we want to participate and to make it a bit more transparent within the team and the community.

I think part of how we can improve our outreach is by having more than just the introductory presentation about Kairos. IMO people rarely come to a talk to know about something new (unless it's already a buzz topic) and see if the given project is a match for their issues. Normally things go the other way around, you have a problem, and you find a presentation where they claim to have a solution for it, and if there's a certain project behind it, then you consider using it.

Ideally, we would love to share Kairos together with someone from the community who is using Kairos, if that is you, please don't hesitate to reach out!

So, I would like to use this discussion to track topics, which, we believe, could be transformed into presentations which then we can pitch at different events.

Here are some of the major topics around Kairos that, I think, are worth a talk:

- Confidential Computing
- Containers
- GitOps
- Kubernetes (obviously :P)
- EdgeVPN
- SBOM

Some other topics which might be a bit more niche but might still be good to share about:

- Immucore
- AuroraBoot

What other topics would you add? And more important, what specific talks do you think could come up from the different topics. Keep in mind that this doesn't need to be limited to traditional presentations, conferences and meetups also offer ways to do workshops and other kind of sharing, so be as creative as you want.

c.c. @kairos-io/maintainers

**Comment by mauromorales** [comment]
Home projects using Kairos could also be interesting to share about

**Comment by Itxaka** [comment]
One interesting demo that I would do is to have a running kairos with an exploitable version of some software. Exploit it, gain privs on it and start whatever you want (like a bitcoin miner lol).

Then just restart the machine and show how everything went away. Even nicer if this is done in a cluster for example and even nicer if we show how after the reset of the cluster it configures itself back to a working state.

Also, now that we have the passive upgrade path, I would also show a demo of recovering the system and how easy it is. Like really breaking up the Active boot somehow and showing how easy is to reset the system to a working state, keeping the system configuration.

Could probably also work a nice demo setting one of those kiosk softwares that run in public places.


Also there should be some demos of use-cases by people, for example, @sarg3nt Kairos with Vault.

**Comment by mauromorales** [comment]
Nice ideas, I think @areitz86 built one of those kiosks

**Reply by areitz86** [reply]
yup, we did. based on fedora. it boots dark, starts straight into chromium, dims up and does some display magic.
we added cillium external workloads to join it to another kairos based server running k3s. so chromium is displaying contents from a service, from the connected cluster, while not running k3s itself ;)

**Comment by jbalonso** [comment]
It's challenging (if not impossible) for me to get out to a conference, but on the matter of "home lab" set ups, I have the following grab bag of features... If any reasonable subset seems interesting enough to discuss, I might be able to put something together:

- Dual stack ipv4/ipv6
- Hybrid arm64 (Raspberry pi) and amd64 (local VMs) cluster
- HA k3s with kube-vip
- fluxcd configuration management
- Longhorn distributed storage (some rpi nodes have m.2 SSDs)
- node-feature-discovery detecting smart home gateways and UPS interfaces (coming someday: 3d printers and octoprint)
- apcupsd daemonsets for emergency cluster shutdown during power outages
- cilium with dual stack BGP peering for publishing LoadBalancer services
- cilium pod-to-pod encrypted traffic with wireguard
- cilium cluster mesh (used during migration from talos to kairos)
- external-dns automated publishing of DNS records for both public and private networks/zones
- step-ca and step-certificates automated CA via cert-manager and ACME
- home-assistant smart home management
- opensprinkler lawn sprinkler system
- self-hosted docker registry
- self-hosted LDAP (not my finest)
- prometheus operator (just getting started)
- self-hosted git repository (gitbucket, though I'll probably migrate to gitea)
- self-hosted vaultwarden (bitwarden clone)

**Reply by jbalonso** [reply]
To add to the above madness, coming up:
- migrating my dockerized ldap-backed kerberos servers into kubernetes
- attempting to move dockerized dhcpd into k8s
- attempting to move my household dns resolver (and internal zone servers) into k8s, navigating circular dependencies for nodes looking for the self-hosted registry
- replacing k8s gitbucket and dockerized Jenkins with k8s gitea and "gitea runners" in k8s

**Reply by mauromorales** [reply]
oh wow, that's awesome @jbalonso 👏 ... for starters, I'd say that sharing one or more of these at one of our meetings would be fun for the team, or reading a blog post about it or whatever you feel more comfortable with
